Store.js

```diff
@@ -140,14 +150,109 @@ define(["dcl/dcl", "decor/Invalidating"], function (dcl, Invalidating) {
 						collection.on("delete", this._itemRemoved.bind(this));
 						collection.on("refresh", this._refreshHandler.bind(this));
 					}
-					return this.processCollection(collection);
+				} else {
+					for (var i = 0; i < this.store.length; i++) {
+						// affect the callback to the observe function if the item is observable
+						Observable.observe(this.store[i], this.observeCallback.bind(this));
+					}
+					collection = processQueryResult.call(this, this.store.filter(this._isQueried, this));
+					// affect the callback to the observe function if the array is observable
+					this._observeResult = ObservableArray.observe(this.store, this.observeCallback.bind(this));
 				}
+				return this.processCollection(collection);
 			} else {
 				this.initItems([]);
 			}
 		},
 
 		/**
+		 * Synchronously deliver change records of the Observe function for the Array
+		 * Do NOT deliver change records of the Observe function for Objects
+		 */
+		deliver: dcl.superCall(function (sup) {
```

Comment:
For the first point it was related to keep Observable.observe returns or not we discussed.
For the second is this function expect to call the remove function of the observe return ?